### DISCUSSION TITLE
Vertical Compaction异常导致集群部分节点崩溃。

### DISCUSSION THREAD

**Discussion by dangerousfeng** [apache/doris#21752] [A - General / Q&A]
be.out如下：
F0712 15:26:36.550199 24721 segment_writer.cpp:746] Check failed: _row_count == _num_rows_written (2767526 vs. 2767584) 
*** Check failure stack trace: ***
    @     0x55915ad7c619  google::LogMessageFatal::~LogMessageFatal()
    @     0x559153dca8ae  doris::segment_v2::SegmentWriter::finalize_columns_data()
    @     0x559153e21a81  doris::VerticalBetaRowsetWriter::_flush_columns()
    @     0x559153e20bb0  doris::VerticalBetaRowsetWriter::add_columns()
    @     0x559153766d7c  doris::Merger::vertical_compact_one_group()
    @     0x559153768306  doris::Merger::vertical_merge_rowsets()
    @     0x559153759a10  doris::Compaction::do_compaction_impl()
    @     0x559153758e3a  doris::Compaction::do_compaction()
    @     0x559153ee024c  doris::BaseCompaction::execute_compact_impl()
    @     0x559153758b13  doris::Compaction::execute_compact()
    @     0x559153f03bce  doris::Tablet::execute_compaction()
    @     0x5591536fa9fe  std::_Function_handler<>::_M_invoke()
    @     0x55915422a0bf  doris::ThreadPool::dispatch_thread()
    @     0x55915421ff5c  doris::Thread::supervise_thread()
    @     0x7f1b7f894ea5  start_thread
    @     0x7f1b800adb0d  __clone
    @              (nil)  (unknown)
*** Query id: 0-0 ***
*** Aborted at 1689146796 (unix time) try "date -d @1689146796" if you are using GNU date ***
*** Current BE git commitID: afe6bb9 ***
*** SIGABRT unknown detail explain (@0x5e1e) received by PID 24094 (TID 24721 OR 0x7f1a01df2700) from PID 24094; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /root/src/doris/be/src/common/signal_handler.h:413
 1# 0x00007F1B7FFE5400 in /lib64/libc.so.6
 2# gsignal in /lib64/libc.so.6
 3# abort in /lib64/libc.so.6
 4# 0x000055915AD825F9 in /data/app/doris-be-2.0.0-beta/lib/doris_be
 5# google::LogMessage::SendToLog() in /data/app/doris-be-2.0.0-beta/lib/doris_be
 6# google::LogMessage::Flush() in /data/app/doris-be-2.0.0-beta/lib/doris_be
 7# google::LogMessageFatal::~LogMessageFatal() in /data/app/doris-be-2.0.0-beta/lib/doris_be
 8# doris::segment_v2::SegmentWriter::finalize_columns_data() in /data/app/doris-be-2.0.0-beta/lib/doris_be
 9# doris::VerticalBetaRowsetWriter::_flush_columns(std::unique_ptr<doris::segment_v2::SegmentWriter, std::default_delete<doris::segment_v2::SegmentWriter> >*, bool) at /root/src/doris/be/src/olap/rowset/vertical_beta_rowset_writer.cpp:126
10# doris::VerticalBetaRowsetWriter::add_columns(doris::vectorized::Block const*, std::vector<unsigned int, std::allocator<unsigned int> > const&, bool, unsigned int) at /root/src/doris/be/src/olap/rowset/vertical_beta_rowset_writer.cpp:108
11# doris::Merger::vertical_compact_one_group(std::shared_ptr<doris::Tablet>, doris::ReaderType, std::shared_ptr<doris::TabletSchema>, bool, std::vector<unsigned int, std::allocator<unsigned int> > const&, doris::vectorized::RowSourcesBuffer*, std::vector<std::shared_ptr<doris::RowsetReader>, std::allocator<std::shared_ptr<doris::RowsetReader> > > const&, doris::RowsetWriter*, long, doris::Merger::Statistics*) in /data/app/doris-be-2.0.0-beta/lib/doris_be
12# doris::Merger::vertical_merge_rowsets(std::shared_ptr<doris::Tablet>, doris::ReaderType, std::shared_ptr<doris::TabletSchema>, std::vector<std::shared_ptr<doris::RowsetReader>, std::allocator<std::shared_ptr<doris::RowsetReader> > > const&, doris::RowsetWriter*, long, doris::Merger::Statistics*) at /root/src/doris/be/src/olap/merger.cpp:350
13# doris::Compaction::do_compaction_impl(long) at /root/src/doris/be/src/olap/compaction.cpp:319
14# doris::Compaction::do_compaction(long) at /root/src/doris/be/src/olap/compaction.cpp:101
15# doris::BaseCompaction::execute_compact_impl() at /root/src/doris/be/src/olap/base_compaction.cpp:87
16# doris::Compaction::execute_compact() at /root/src/doris/be/src/olap/compaction.cpp:83
17# doris::Tablet::execute_compaction(doris::CompactionType) at /root/src/doris/be/src/olap/tablet.cpp:1825
18# std::_Function_handler<void (), doris::StorageEngine::_submit_compaction_task(std::shared_ptr<doris::Tablet>, doris::CompactionType, bool)::$_0>::_M_invoke(std::_Any_data const&) at /var/local/ldb-toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:291
19# doris::ThreadPool::dispatch_thread() in /data/app/doris-be-2.0.0-beta/lib/doris_be
20# doris::Thread::supervise_thread(void*) at /root/src/doris/be/src/util/thread.cpp:466
21# start_thread in /lib64/libpthread.so.0
22# clone in /lib64/libc.so.6

目前be.conf: 
enable_vertical_compaction = false
关闭vertical_compaction后不会出现崩溃，是否vertical_compaction存在问题？

**Comment by yagagagaga** [comment]
This maybe a vulnerabilities, maybe we can add a WeChat and have a deeper communication.
https://github.com/apache/doris/discussions/16887#discussioncomment-5165839
